File: main/coreplugins/io.sarl.intellij/src/main/java/io/sarl/intellij/SarlFileType.java
package io.sarl.intellij;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class SarlFileType extends LanguageFileType {
    public static final SarlFileType INSTANCE = new SarlFileType();

    protected SarlFileType() {
        super(SarlLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Sarl";
    }

    @NotNull
    @Override
    public String getDescription() {
        // TODO return SarlFileType.message("sarl.file.type.description");
        return "Sarl file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "sarl";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return SarlIcons.SARL_PLUGIN;
    }
}

File: main/coreplugins/io.sarl.intellij/src/main/kotlin/io/sarl/intellij/SarlFileType.kt
package io.sarl.intellij

import com.intellij.openapi.fileTypes.LanguageFileType

import javax.swing.*

object SarlFileType : LanguageFileType(SarlLanguage.INSTANCE) {

    override fun getName(): String {
        return "Sarl"
    }

    override fun getDescription(): String {
        return "Sarl Files"
    }

    override fun getDefaultExtension(): String {
        return DEFAULTS.EXTENSION
    }

    override fun getIcon(): Icon? {
        return SarlIcons.SARL_PLUGIN
    }

    object DEFAULTS {
        const val EXTENSION = "sarl"
    }
}